File: 2160.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int minimumSum(int num) {
        vector<int> temp;
        while (num != 0) {
            temp.emplace_back(num % 10);
            num /= 10;
        }        
        while (temp.size() != 4) {
            temp.emplace_back(0);
        }
        sort(temp.begin(), temp.end());
        return temp[0] * 10 + temp[2] + temp[1] * 10 + temp[3];
    }
};

File: 2160.go
package leetcode

import "sort"

func minimumSum(num int) int {
	var temp []int
	for num != 0 {
		temp = append(temp, num%10)
		num /= 10
	}
	for len(temp) < 4 {
		temp = append(temp, 0)
	}
	sort.Ints(temp)
	return temp[0] * 10 + temp[2] + temp[1] * 10 + temp[3]
}